board/batocera/fsoverlay/etc/init.d/S99userservices

```diff
@@ -17,28 +17,38 @@ then
 fi
 
 # user services
-find /userdata/system/services -type f |
+find -L /userdata/system/services -type f |
     while read SERVICE
     do
-	BASE_SERVICE=${SERVICE##*/}
-	SRVVAR=__SERVICE__${BASE_SERVICE%.*}
-	if test "${!SRVVAR}" = 1
-	then
-	    bash "${SERVICE}" "${1}" &
-	fi
+    BASE_SERVICE=${SERVICE##*/}
+    if echo "${BASE_SERVICE,,}" | grep -q -E ^[a-z].[a-z0-9_]*$; then
+        SRVVAR=__SERVICE__${BASE_SERVICE}
+        if test "${!SRVVAR}" = 1; then
+    	    bash "${SERVICE}" "${1}" &
+            echo "User Service: ${BASE_SERVICE} - service is enabled - filename [ok]"
+        else
+            echo "User Service: ${BASE_SERVICE} - service is disabled - filename [ok]"            
+        fi
+    else
+        string_new=${BASE_SERVICE//[^A-Za-z0-9_]/}
+        string_new=$(echo "$string_new" | sed 's/[[:digit:]]*//')
```

Comment:
This is wrong, it will remove the first string of digits contained in the string, instead of removing them only from the front of the string.  Anchor with `^`

Reply:
No, this is not a reg-expression you may think
it negates the content in square brackets
test=abcdefg ${test//abc/} --> defg
test=abcdefg ${test//[^abc]/} --> abc

Reply:
My comment pertains to line 34.  I'm quite sure it's a regular expression, and that it's wrong.  For a test case, create a script GrüßGott12345678abc and show the recommendation of your script.